_build-commons.sh

```diff
@@ -0,0 +1,462 @@
+#!/usr/bin/env bash
```

Comment:
I don't like the fact that we have now combined command line options for build.sh and built-tests.sh into one common set. Half of the options doesn't make sense for build.sh and half of them doesn't make sense for build-tests.sh. 
I would prefer if we parsed only the common options here and parsed the options that belong just to build.sh or build-test.sh in the respective .sh files. Maybe we can let the common parsing code call a callback function for each parameter it doesn't know. Then build.sh / build-test.sh would pass in its own function that would parse their specific parameters.

Reply:
Also, we print the list of supported options when you pass --help option in the `usage` function. It seems that the combined options description should come from this file. Otherwise adding a common option would require updating the help at two places.

Reply:
Added Callback function `handle_arguments`. Basically it is order-dependent (appears before the `source _build-commons.sh`), like the rest of the variable/function declarations.

Added `usage_list` in caller scripts, that gets printed unified `usage()` function as Additional Options. There were some desyncs which I fixed along the way; some options are common, but missing from original `build-test.sh`, either in case-esac or in usage or both (e.g. numProc, `skipconfigure` etc.). Therefore, the common list might have few extras cases, which are not in the original (master branch) intersection.